### PR TITLE
Improve legibility of path regular expressions.

### PR DESCRIPTION
**Pull request by sashadoes**
## Summary

Addresses issue:

- #5228 

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics/components/settings/SettingsEdit.test.js [hunk 1]
```diff
@@ -58,10 +58,12 @@ describe( 'SettingsEdit', () => {
...
 		] );
 
-		fetchMock.get( /analytics\/data\/settings/, { body: {} } );
-		fetchMock.get( /analytics-4\/data\/settings/, { body: {} } );
+		fetchMock.get( new RegExp( 'analytics/data/settings' ), { body: {} } );
+		fetchMock.get( new RegExp( 'analytics-4/data/settings' ), {
+			body: {},
+		} );
 
-		fetchMock.get( /\example.com/, {
```
Comment (tofumatt) [inline assets/js/modules/analytics/components/settings/SettingsEdit.test.js:64]: Odd, seems like that was actually an error; the `\` here seems unneeded. Thanks for catching that! 👍🏻

**Review by tofumatt — CHANGES_REQUESTED** [review]
Overall looks good, just a few questions/comments and then should be good to merge. Thanks for this one! 👍🏻

**Comment by sashadoes** [comment]
@tofumatt Thanks for your review and feedback. 
I made an update:)

File: packages/eslint-plugin/rules/jsdoc-third-person.js [hunk 2]
```diff
@@ -51,7 +51,7 @@ module.exports = iterateJsdoc(
 		// Verbs can include dashes or in some cases also parentheses.
 		if (
 			jsdoc.description &&
-			! jsdoc.description.match( /^[\w\(\)\-]+s\W.*/g )
+			! jsdoc.description.match( /^[\w\(\)\-]+s\W.*/g ) // eslint-disable-line
```
Comment (tofumatt) [inline packages/eslint-plugin/rules/jsdoc-third-person.js:54]:
```suggestion
			! jsdoc.description.match( /^[\w\(\)\-]+s\W.*/g ) // eslint-disable-line
```

Why is this line being disabled? As mentioned in this comment: https://github.com/google/site-kit-wp/pull/6295#discussion_r1044676281, we should be specific about rules we disable, and why.

I think here we should be able to convert this to a `new RegExp` format, right?

**Review by tofumatt — CHANGES_REQUESTED** [review]
There's one `eslint-disable` here that we can probably remove, then this should be good.

**Review by tofumatt — APPROVED** [review]

File: assets/js/components/settings/SettingsActiveModule/ConfirmDisconnect.js [hunk 3]
```diff
@@ -153,7 +153,7 @@ export default function ConfirmDisconnect( { slug } ) {
 				'google-site-kit'
 			),
 			name,
-			listFormat( dependentModules )
```
Comment (aaemnnosttv) [inline assets/js/components/settings/SettingsActiveModule/ConfirmDisconnect.js:156]: Why was this removed?

File: assets/js/modules/analytics/components/settings/SettingsEdit.test.js (hunk 1 above)
Comment (aaemnnosttv) [reply assets/js/modules/analytics/components/settings/SettingsEdit.test.js:64]: Shouldn't the `.` be escaped in `.com`?

**Review by aaemnnosttv — COMMENTED** [review]
⚠️ Looks like there was one thing changed here which was unrelated to the requirements.

File: assets/js/components/settings/SettingsActiveModule/ConfirmDisconnect.js (hunk 3 above)
Comment (tofumatt) [reply assets/js/components/settings/SettingsActiveModule/ConfirmDisconnect.js:156]: Huh, that's odd, not sure? I missed that in the huge PR, my bad 😞

File: assets/js/modules/analytics/components/settings/SettingsEdit.test.js (hunk 1 above)
Comment (tofumatt) [reply assets/js/modules/analytics/components/settings/SettingsEdit.test.js:64]: It should have been, yes. But of course it'll still detect `example.com` with `.` operating as a wildcard instead.

It'll actually need to be `new RegExp( 'example\\.com' )`.

File: assets/js/components/settings/SettingsActiveModule/ConfirmDisconnect.js (hunk 3 above)
Comment (tofumatt) [reply assets/js/components/settings/SettingsActiveModule/ConfirmDisconnect.js:156]: Might've been a bad merge actually, looking at things. 🤔 Can't see a reason why it'd be removed otherwise. 🤷🏻‍♂️
